### PR TITLE
Sa annotations

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Add ability to customize ServiceAccount annotations

# Context

AWS EKS IAM service accounts requires adding annotations to the service account.
https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-technical-overview.html

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2698

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick e2e test flake, bind address in use https://console.cloud.google.com/cloud-build/builds/39e06292-6403-4c97-80cf-2b30c55dee61;step=8?project=solo-public

File: install/helm/gloo/templates/2-gloo-service-account.yaml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
+{{- if .Values.gloo.serviceAccount.extraAnnotations }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/2-gloo-service-account.yaml:4]: any reason we're doing gloo, discovery, and gateway-proxy here but not adding extra annotations to gateway, or even have a service account for knative/ingress proxy?

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -156,17 +158,22 @@ type DiscoveryDeployment struct {
...
+	Deployment                    *GatewayDeployment  `json:"deployment,omitempty"`
+	CertGenJob                    *CertGenJob         `json:"certGenJob,omitempty" desc:"generate self-signed certs with this job to be used with the gateway validation webhook. this job will only run if validation is enabled for the gateway"`
+	UpdateValues                  bool                `json:"updateValues" desc:"if true, will use a provided helm helper 'gloo.updatevalues' to update values during template render - useful for plugins/extensions"`
+	ProxyServiceAccount           ProxyServiceAccount `json:"proxyServiceAccount" `
+	ReadGatewaysFromAllNamespaces bool                `json:"readGatewaysFromAllNamespaces" desc:"if true, read Gateway custom resources from all watched namespaces rather than just the namespace of the Gateway controller"`
 }
 
-type ServiceAccount struct {
+type ProxyServiceAccount struct {
 	DisableAutomount bool `json:"disableAutomount" desc:"disable automunting the service account to the gateway proxy. not mounting the token hardens the proxy container, but may interfere with service mesh integrations"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go:171]: should we just be adding this to all our service accounts?

File: test/e2e/aws_test.go [hunk 3]
```diff
@@ -315,7 +315,7 @@ var _ = Describe("AWS Lambda", func() {
 		It("should be able to call lambda via gateway", testLambdaWithVirtualService)
 	})
 
-	Context("Temporary Credentials", func() {
+	FContext("Temporary Credentials", func() {
```
Comment (kdorosh) [inline test/e2e/aws_test.go:318]: unfocus

File: changelog/v1.5.0-beta14/sa-extra-annotations.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: NEW_FEATURE
```
Comment (kdorosh) [inline changelog/v1.5.0-beta14/sa-extra-annotations.yaml:2]:
```suggestion
- type: HELM
```

File: test/e2e/aws_test.go [hunk 5]
```diff
@@ -324,7 +324,7 @@ var _ = Describe("AWS Lambda", func() {
 				Skip("no AWS creds available")
 			}
 			stsClient := sts.New(sess)
-
+			stsClient.AssumeRoleWithWebIdentityRequest()
```
Comment (kdorosh) [inline test/e2e/aws_test.go:327]: why this addition?

Comment (EItanya) [reply test/e2e/aws_test.go:327]: ahhh good point
local mistake

**Pull request merged**